sympy/polys/polytools.py

```diff
@@ -6218,8 +6218,20 @@ def sqf_list(f, *gens, **args):
     (2, [(x + 1, 2), (x + 2, 3)])
 
     """
-    return _generic_factor_list(f, gens, args, method='sqf')
-
+    options.allowed_flags(args, ['frac', 'polys'])
+    opt = options.build_options(gens, args)
+    f = sympify(f)
+    if isinstance(f, Expr) and not f.is_Relational:
+        numer, denom = together(f).as_numer_denom()
```

Comment:
This is not needed as there should be no denominator. It there is a denominator, the polynomial creation methods will raise PolynomialError.

Reply:
`_poly_from_expr(f, opt)` is allowing denominator to be added to the polynomial. For example:

`sqf_list(  (x**2 + 1)  * (x - 1)**2 * (x - 2)**3 * (x - 3)**3 /(x + 2))`

this generates a polynomial with two generators, `1/(x+2)` and `x`

Reply:
We should check that there is only one generator and that it is a Symbol.